conans/client/hook_manager.py

```diff
@@ -19,6 +19,37 @@ def pre_export(output, conanfile, conanfile_path, reference, **kwargs):
                         % field)
 """
 
+recipe_typo_checker_hook = """
+import inspect
+from difflib import get_close_matches
+
+from conans import ConanFile
+
+
+def pre_export(output, conanfile, conanfile_path, reference, **kwargs):
+    def get_members(conanfile):
+        return [m[0] for m in inspect.getmembers(conanfile) if not m[0].startswith('_')]
+
+    base_members = get_members(ConanFile)
+    base_members.extend(["requires", "build_requires", "requirements",
+                         "build_requirements", "python_requires", "keep_imports",
+                         "imports", "build_id", "deploy", "scm", "conan_data",
+                         "display_name", "output"])
```

Comment:
There is a lot of stuff to be added manually. Is it possible to expose at least some of it to the base recipe?